airbyte-integrations/connectors/source-zendesk-sell/source_zendesk_sell/source.py

```diff
@@ -0,0 +1,301 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+import re
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams import IncrementalMixin
+
+# Basic full refresh stream
+class ZendeskSellStream(HttpStream, ABC):
+    """
+    This class represents a stream output by the connector.
+    This is an abstract base class meant to contain all the common functionality at the API level e.g: the API base URL, pagination strategy,
+    parsing responses etc..
+    """
+
+    url_base = "https://api.getbase.com/v2/"
+    primary_key = None
+
+    # def __init__(self, config: Mapping[str, Any], **kwargs):
+    #     super().__init__()
+    #     self.start_date = config["start_date"]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        try:
+            regex_page='[=?/]page[_=/-]?(\d{1,3})'
+            meta_links = response.json().get('meta', {}).get('links')
+            if 'next_page' in meta_links.keys():
+                return {'page' : int(re.findall(regex_page, meta_links['next_page'])[0])}
+        except Exception as e:
+            print(f"{e.__class__} occurred, while trying to get next page information from the following dict {meta_links}") 
+            return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        if next_page_token != None:
+            return {'page': next_page_token['page']}
+        else:
+            return {}
```

Comment:
@alexchouraki usually the start_date or state is added in `request_params` here adding.

Check how is made in Zendesk Chat connector:
https://github.com/airbytehq/airbyte/blob/8c3162aeba345667510f1f8d538eca26fa18339f/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py#L125-L139

Reply:
Zendesk Chat seems to support queries using a start_date param : https://developer.zendesk.com/api-reference/live-chat/chat-api/incremental_export/
As far as I understand, it's not the case for Zendesk Sell : https://developer.zendesk.com/api-reference/sales-crm/resources/contacts/#retrieve-all-contacts. There is no equivalent, only a field I can use to "sort-by". I could sort by updated_at, but if I have a parameter like "start_time" in my request, I'll just get 400 bad requests errors...

Reply:
So I'd need to have a start_time parameter somewhere in my config, sort by updated_at, retrieve everything, and then throw out the unnecessary data?

Reply:
not necessarily because when you update the header request it will request only data started from that date.

Reply:
Again, I don't understand how that would work if the API is not able to handle it.
Doing a simple request to Zendesk Sell with a parameter like this results in a 400 error.
<img width="908" alt="Capture d’écran 2022-10-19 à 11 37 27" src="https://user-images.githubusercontent.com/32514048/196655347-b0c66757-13ab-4ff9-ae39-a1891fc90c48.png">
Same thing with start_time, or whatever. Once again, the API doesn't support [it.](https://developer.zendesk.com/api-reference/sales-crm/resources/contacts/#retrieve-all-contacts), and throws a 400 error whenever there is a parameter like this involved.

So please let me know if :
- I'm missing something, and it's actually possible to do incremental with this REST API, in which case I'd love to have an example of params that allow to request only data started from a given date, to understand how to actually implement it
- I need to go away from the REST API, in which case I'd need to work with Zendesk Sell's Search API as well, which is limited to 3 endpoints. Maybe I can use the search API for those 3 endpoints and have them incremental and taking into account a start_date, and the rest API for the others, that won't be incremental at all ?